### DISCUSSION TITLE
ImportError: 'NeighborSampler' requires either 'pyg-lib' or 'torch-sparse'

### DISCUSSION THREAD

**Discussion by hmam9090** [pyg-team/pytorch_geometric#9303] [Q&A]
I'm learning Pytorch_geometric and I ran into this error. Im using Ubuntu-22.04 and only CPU. I Have pyg-lib installed already and torch-sparse but Im still getting this error. 
![image](https://github.com/pyg-team/pytorch_geometric/assets/128145688/f05370f8-6379-473a-a0f1-ef69a4a63ce9)
HEre is a list of my packages: aiohttp            3.9.5
aiosignal          1.3.1
async-timeout      4.0.3
attrs              23.2.0
Brotli             1.0.9
certifi            2024.2.2
cffi               1.16.0
charset-normalizer 2.0.4
filelock           3.13.1
frozenlist         1.4.1
fsspec             2024.3.1
gmpy2              2.1.2
idna               3.7
Jinja2             3.1.3
joblib             1.4.2
MarkupSafe         2.1.3
mkl-fft            1.3.8
mkl-random         1.2.4
mkl-service        2.4.0
mpmath             1.3.0
multidict          6.0.5
networkx           3.1
numpy              1.26.4
pillow             10.3.0
pip                23.3.1
psutil             5.9.8
pycparser          2.21
pyg-lib            0.4.0+pt21cpu
pyparsing          3.1.2
PySocks            1.7.1
PyYAML             6.0.1
requests           2.31.0
scikit-learn       1.4.2
scipy              1.13.0
setuptools         69.5.1
six                1.16.0
sympy              1.12
threadpoolctl      3.5.0
torch_cluster      1.6.3
torch_geometric    2.5.3
torch_scatter      2.1.2
torch_sparse       0.6.18
torch_spline_conv  1.2.2
torchaudio         2.1.0
torchvision        0.16.0
tqdm               4.66.4
typing_extensions  4.9.0
urllib3            2.1.0
wheel              0.43.0
yarl               1.9.4

**Comment by rusty1s** [comment]
What does
```python
import torch_geometric

print(torch_geometric.typing.WITH_PYG_LIB)
print(torch_geometric.typing.WITH_TORCH_SPARSE)
```
return? Does
```python
import pyg_lib
import torch_sparse
```
work for you?

**Reply by hmam9090** [reply]
I have managed to fix it. I had to create a fresh env and use poetry to install everything!
